crates/execution/src/pipelined.rs

```diff
@@ -109,6 +152,53 @@ impl ProjectListExecutor {
     }
 }
 
+/// An executor for a query that returns rows from a view.
+/// Essentially just a projection that drops the view's metadata columns.
+#[derive(Debug)]
+pub struct ViewProject {
+    num_cols: usize,
+    num_private_cols: usize,
+    inner: PipelinedProject,
+}
+
```

Comment:
I like a comment here that explain the deal with `num_private_cols`, that is used for filter out the injected ones, also a link to `TableSchema::from_view_def_for_datastore` in both directions that this  need to hold